### PR TITLE
[flang][cuda] Add a proper TODO for allocate statement for cuda var

### PR DESCRIPTION
**Pull request by clementval**
Allocate statement for variable with CUDA attributes need to allocate memory on the device and not the host. Add a proper TODO so we keep track of work to be done for it.

### CODE REVIEW

File: flang/lib/Lower/Allocatable.cpp [hunk 1]
```diff
@@ -379,6 +379,9 @@ class AllocateStmtHelper {
   }
 
   void lowerAllocation(const Allocation &alloc) {
+    if (HasCUDAAttr(alloc.getSymbol()))
```
Comment (jeanPerier) [inline flang/lib/Lower/Allocatable.cpp:382]: don't we need `Fortran::semantics::` here? Obviously, there is probably some using ... spilling here. But unless it is consistently not used in the file, please add it.

File: flang/include/flang/Semantics/tools.h [hunk 2]
```diff
@@ -212,6 +212,16 @@ inline bool IsCUDADeviceContext(const Scope *scope) {
   return false;
 }
 
+inline bool HasCUDAAttr(const Symbol &sym) {
+  if (const auto *details =
+          sym.GetUltimate().detailsIf<semantics::ObjectEntityDetails>()) {
```
Comment (jeanPerier) [inline flang/include/flang/Semantics/tools.h:217]:
```suggestion
  if (const auto *details{
          sym.GetUltimate().detailsIf<semantics::ObjectEntityDetails>()}) {
```

Suggestion probably needs formatting though.

**Review by jeanPerier — APPROVED** [review]
Nits, thanks for adding the TODO

File: flang/include/flang/Semantics/tools.h (hunk 2 above)
Comment (clementval) [reply flang/include/flang/Semantics/tools.h:217]: Thanks! I always miss the brace init switching between different code base rule!

**Review by vzakhari — APPROVED** [review]

File: flang/lib/Lower/Allocatable.cpp (hunk 1 above)
Comment (clementval) [reply flang/lib/Lower/Allocatable.cpp]: It somehow works without it but I added the namespaces to be consistent with the rest of the file.

**Pull request merged**